src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -550,6 +550,29 @@ public void EmptyEnumStoreTest()
             Assert.True(enums.Count() > 0);
         }
 
+        [Fact]
+        public async void MaxRecursionDepthTest()
+        {
+            bool test;
+            var config = new PowerFxConfig(null)
+            {
+                MaxCallDepth = 5
+            };
+            var recalcEngine = new RecalcEngine(config);
+            try

```

Comment:
>try [](http://example.com/codeflow?start=12&length=3)

nit:
Can you do
Assert.Throws<MaxCallDepthException>  ( () => ...

Reply:
That might be better than the boolean assert

Reply:
No longer necessary as I implemented the top level catch at EvalAsync to instead return an Error.